source/languageHandler.py

```diff
@@ -177,6 +172,66 @@ def setLanguage(lang):
 		curLang="en"
 	# #9207: Python 3.8 adds gettext.pgettext, so add it to the built-in namespace.
 	trans.install(names=["pgettext"])
+	setLocale(curLang)
+
+
+def setLocale(localeName):
+	'''
+	Set python's locale using a `localeName` set by `setLanguage`.
+	Python 3.8's locale system allows you to set locales that you cannot get
+	so we must test for both ValueErrors and locale.Errors
+	'''
+
+	r'''
+	>>> import locale
+	>>> locale.setlocale(locale.LC_ALL, 'foobar')
+	Traceback (most recent call last):
+	File "<stdin>", line 1, in <module>
+	File "Python38-32\lib\locale.py", line 608, in setlocale
+		return _setlocale(category, locale)
+	locale.Error: unsupported locale setting
+	>>> locale.setlocale(locale.LC_ALL, 'en-GB')
+	'en-GB'
+	>>> locale.getlocale()
+	Traceback (most recent call last):
+	File "<stdin>", line 1, in <module>
+	File "Python38-32\lib\locale.py", line 591, in getlocale
+		return _parse_localename(localename)
+	File "Python38-32\lib\locale.py", line 499, in _parse_localename
+		raise ValueError('unknown locale: %s' % localename)
+	ValueError: unknown locale: en-GB
+	'''
+	# Try setting Python's locale to lang
+	localeChanged = False
+	try:
+		locale.setlocale(locale.LC_ALL, localeName)
+		locale.getlocale()
+		localeChanged = True
+	except (locale.Error, ValueError):
+		pass

```

Comment:
I think it would be wise to have at least a debug level log message for each of these exception handling blocks. Including the value that was tried.